python-client/giskard/models/model_explanation.py

```diff
@@ -31,13 +48,53 @@ def _get_background_example(df: pd.DataFrame, feature_types: Dict[str, str]) ->
 
 
 def _get_columns_original_order(prepared_dataset: Dataset, model: BaseModel, dataset: Dataset) -> list:
-    """Get columns of the prepared_dataset in the original order."""
+    """Return columns of the `prepared_dataset` in the original order.
+
+    Use `model` or `dataset` to infer columns' original order, which `model` was
+    trained with.
+
+    Parameters
+    ----------
+    prepared_dataset : giskard.Dataset
+        Dataset, which entries need to be explained using SHAP.
+
+    model : giskard.Model
+        Model object, which potentially contains columns order information.
+
+    dataset: giskard.Dataset
+        Reference dataset, which potentially contains columns order
+        information.
+
+    Returns
+    -------
+    list
+        Original order of columns, which `model` was trained on.
+    """
     features_names = model.meta.feature_names
     return features_names if features_names else [c for c in dataset.df.columns if c in prepared_dataset.df.columns]
 
 
 def _prepare_for_explanation(input_df: pd.DataFrame, model: BaseModel, dataset: Dataset) -> pd.DataFrame:
-    """Prepare dataframe for an inference step."""
+    """Prepare dataframe for a SHAP explanation.
+
+    Check necessary columns' presence, data types and order.
```

Comment:
Check the dataframe columns and their order, and cast columns to their data types.

Reply:
\+

Reply:
\+